core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java

```diff
@@ -174,11 +185,16 @@ private void validateVolumeCount(VolumeTemplate value, VmType vmType, VolumePara
     }
 
     private void validateVolumeSize(VolumeTemplate value, VmType vmType, VolumeParameterType volumeParameterType,
-        ValidationResult.ValidationResultBuilder validationBuilder) {
+        ValidationResult.ValidationResultBuilder validationBuilder, InstanceGroup instanceGroup) {
         if (vmType != null && needToCheckVolume(volumeParameterType, value.getVolumeCount())) {
             VolumeParameterConfig config = vmType.getVolumeParameterbyVolumeParameterType(volumeParameterType);
             if (config != null) {
-                if (value.getVolumeSize() > config.maximumSize()) {
+                if (isIDBrokerInstanceGroup(instanceGroup)) {
+                    if (value.getVolumeSize() != 0) {
+                        // IDBroker does not use data volume, so its volume size should be zero
+                        validationBuilder.error(String.format("IDBroker volume size: %s is not zero", value.getVolumeSize()));
+                    }
+                } else if (value.getVolumeSize() > config.maximumSize()) {
                     validationBuilder.error(String.format("Max allowed volume size for '%s': %s", vmType.value(), config.maximumSize()));
                 } else if (value.getVolumeSize() < config.minimumSize()) {
```

Comment:
Like above, I wonder if forcing the size & count to zero causes any troubles for custom DL clusters.